embulk-core/src/main/java/org/embulk/spi/PageBuilder.java

```diff
@@ -170,49 +146,76 @@ public void setTimestamp(int columnIndex, Timestamp value)
     {
         if (value == null) {
             setNull(columnIndex);
-            return;
         }
+        else {
+            row.setTimestamp(columnIndex, value);
+        }
+    }
 
-        int offset = getOffset(columnIndex);
-        bufferSlice.setLong(offset, value.getEpochSecond());
-        bufferSlice.setInt(offset + 8, value.getNano());
+    void writeNull(int columnIndex)
```

Comment:
Any reasons why they are not `private`?

It's okay named `write*` if it is just `private`. But if the methods are somewhat visible, I'm concerned that the difference is not clear between `set*` and `write*`. Please make them `private`, or rename them so that their names represent what they do and/or comment why they are not `private`.